calliope/test/common/test_model/model_demand_share.yaml

```diff
@@ -0,0 +1,76 @@
+model:
+    name: Demand share constraint test model
+    timeseries_data_path: 'timeseries_data'
+
+run:
+    mode: plan
+    solver: glpk
+
+techs:
+    cheap_supply:
+        essentials:
+            name: Cheap supply tech
+            carrier: electricity
+            parent: supply
+        constraints:
+            lifetime: 25
+            energy_cap_max: .inf
+        costs:
+            monetary:
+                interest_rate: 0.1
+                energy_cap: 10
+    expensive_supply:
+        essentials:
+            name: Expensive supply
+            carrier: electricity
+            parent: supply
+        constraints:
+            lifetime: 25
+            energy_cap_max: .inf
+        costs:
+            monetary:
+                interest_rate: 0.1
+                energy_cap: 20
+    electricity_demand:
+        essentials:
+            name: Electricity demand
+            carrier: electricity
+            parent: demand
+        constraints:
+            resource: file=demand_elec.csv
+
+locations:
+    0:
+        techs:
+            cheap_supply:
+            expensive_supply:
+            electricity_demand:
+    1:
+        techs:
+            cheap_supply:
+            expensive_supply:
+            electricity_demand:
+
+overrides:
```

Comment:
Am I right in saying that there's no actual test for a list with length greater than one in either 'locs' or 'techs'? I would maybe have a model with three locations and three techs, so that the inclusion of two of those three within a constraint can be checked.

To ensure the run times don't get excessive, it is probably sufficient to build the model with pyomo, but not run it. The constraint can still be checked to exist (`constraint_exists` function) and the upper/lower bound can be checked (e.g. `test_backend_pyomo.py` line 488).